src/platforms/linux/daemon/iputilslinux.cpp

```diff
@@ -56,7 +61,10 @@ bool IPUtilsLinux::setMTUAndUp() {
   strncpy(ifr.ifr_name, WG_INTERFACE, IFNAMSIZ);
 
   // MTU
-  ifr.ifr_mtu = 1420;
+  // FIXME: We need to know how many layers deep this particular
```

Comment:
Follow up?

Reply:
Good catch! This is mostly a case of the comment being stale after some of the refactoring that went on in this PR. Since this one interface is shared by all hops, we have to set an MTU at least big enough for all the hops. Assuming the route to the internet has an Ethernet MTU of 1500, then the correct value to choose would be `ETH_MTU - WG_MTU_OVERHEAD`.

However, the spirit behind this comment is that each hop adds another `WG_MTU_OVERHEAD` to the packet, meaning that the final hops have a smaller effective MTU than the first hops. This isn't something that we can configure on the interface and we would have to do it in the routing table somehow.